### DISCUSSION TITLE
Directly trigger button event inside IView

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#1035] [Q&A]
Hi,
    I want to click a button and trigger a click event on that button. the button is inside a custom snabbdom view, code as follow
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/179aef53-ba92-4a1b-a771-ad290df8dafd)
![image](https://github.com/eclipse-glsp/glsp/assets/4537809/da00817b-0356-4fc8-b003-cd6bebd812f8)

my problem is that, if I direct click the button, click event will not be triggered, I should select the view first then click the button can trigger the event.
![2023-06-16_17-18-29](https://github.com/eclipse-glsp/glsp/assets/4537809/5add4ac0-865f-46b2-8c7a-15662a9e85b1)
Could the click event be triggered when clicking the button?

**Comment by martin-fleck-at** [comment]
@wss29 That is a good question! I've never tried it but it seems that other people had similar issues with svg+foreignObject and resolved them through, e.g., "svg-native" click zones (e.g., https://github.com/software-mansion/react-native-svg/issues/1389). In theory [foreignObject should support onClick](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/onclick) without issues but maybe it does not properly propagate the event. Can you see whether the click event is fired on the foreignObject if you add a listener there?

On a "regular" web page it definitely works so there might be some forwarding issue:
```
<!DOCTYPE html>
<svg style="width: 100px; height: 100px; background: yellow">
  <foreignObject style="transform: translateY(20px); width: 100px; height: 100px; background: lime">
    <button onClick="alert('test')">Test</button>
  </foreignObject>
</svg>
```

**Comment by martin-fleck-at** [comment]
FYI, I just did a test with this IView as part of the workflow example:

```
import { IView, IViewArgs, on, RenderingContext, SShapeElement, svg } from '@eclipse-glsp/client';
import { injectable } from 'inversify';
import { VNode } from 'snabbdom';
import virtualizeString from 'sprotty/lib/lib/virtualize';

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const JSX = { createElement: svg };

@injectable()
export class CustomShape implements IView {
    render(node: Readonly<SShapeElement>, _context: RenderingContext, _args?: IViewArgs): VNode | undefined {
        const div = document.createElement('div');
        div.className = 'container';
        div.style.padding = '20px';
        div.style.backgroundColor = 'green';

        const button = document.createElement('button');
        button.name = node.id;
        button.id = node.id;
        button.textContent = 'Click';

        div.appendChild(button);

        const foreignVNode = virtualizeString(div.outerHTML);
        on(foreignVNode!, 'click', (event: any) => console.log('CLICKED'));
        const vnode: VNode = (
            <g>
                <foreignObject requiredFeatures='http://www.w3.org/TR/SVG11/feature#Extensibility' height={50} width={70} x={0} y={0}>
                    {foreignVNode}
                </foreignObject>
            </g>
        );
        return vnode;
    }
}
```

![image](https://github.com/eclipse-glsp/glsp/assets/19170971/3c28c9d3-cdc1-4f20-a5f0-7083689b1837)


And for me the click is properly triggered and logged to the console.
